src/create-probot-app.ts

```diff
@@ -2,13 +2,14 @@
 import { askUser, runCliManager } from "./helpers/user-interaction";
 import { initGit } from "./helpers/init-git";
 import { installAndBuild } from "./helpers/run-npm";
-import { makeScaffolding } from "./helpers/filesystem";
+import { ensureValidDestination, makeScaffolding } from "./helpers/filesystem";
 import { printSuccess, red } from "./helpers/write-help";
 
 async function main(): Promise<void> {
   const config = await askUser(runCliManager());
 
-  makeScaffolding(config)
+  ensureValidDestination(config.destination, config.overwrite)
```

Comment:
The inquirer prompts will be asked, and then an error will be thrown at the end, if the destination was invalid. We should validate the destination as early as possible.
Could be done by moving the `ensureValidDestination` check to `user-interation.ts`. as shown below.


```diff
diff --git a/src/helpers/user-interaction.ts b/src/helpers/user-interaction.ts
index cc135e6..5aca32b 100644
--- a/src/helpers/user-interaction.ts
+++ b/src/helpers/user-interaction.ts
@@ -218,6 +218,7 @@ export function runCliManager(): CliConfig {
   const options = program.parse(process.argv).opts();
 
   if (!destination) printHelpAndFail();
+  if(!ensureValidDestination()) printDesinationHelpOrWhateverAndFail();
   if (options.showTemplates) {
     getTemplates().forEach((template) => console.log(template));
     process.exit();
```